### PR TITLE
added support for --set-gtid-purged argument

### PR DESCRIPTION
**Pull request by necenzurat**
Fixes: https://github.com/spatie/laravel-backup/issues/625

### CODE REVIEW

File: src/Databases/MySql.php [hunk 1]
```diff
@@ -100,6 +103,16 @@ public function setDefaultCharacterSet(string $characterSet)
         return $this;
     }
 
+    /**
+     * @return $this
+     */
+    public function setGtidPurged()
```
Comment (freekmurze) [inline src/Databases/MySql.php]: Can this be named better?

Seems confusing from the outside that this method name is named positively, while the option is something negative `--set-gtid-purged=OFF`. 

Something like `gtidWasPurged` sounds better to me.

Comment (freekmurze) [reply src/Databases/MySql.php]: Feel free to propose something even better.

Comment (necenzurat) [reply src/Databases/MySql.php]: Looking back on iit, it has 3 states, ON, OFF and AUTO. [Source](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html#option_mysqldump_set-gtid-purged).
Will try to name it properly and add the rest of the functionality.

**Comment by necenzurat** [comment]
Updated, with whole functionality for ```--set-gtid-purged=ON/OFF/AUTO```, but cannot find a suitable name for it.
But now ```setGtidPurged``` it kinda makes sense.

File: src/Databases/MySql.php [hunk 2]
```diff
@@ -182,6 +195,10 @@ public function getDumpCommand(string $dumpFile, string $temporaryCredentialsFil
             $command[] = $extraOption;
         }
 
+        if ($this->setGtidPurged != 'AUTO') {
```
Comment (freekmurze) [inline src/Databases/MySql.php]: Make this a strict comparison `!==`

Comment (necenzurat) [reply src/Databases/MySql.php]: added

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**